### PR TITLE
fix: ChannelManager concurret Release and Watch bug

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #38589

### CODE REVIEW

**Comment by czs007** [comment]
rerun go-sdk

File: internal/datacoord/channel.go [hunk 1]
```diff
@@ -191,7 +191,15 @@ func NewStateChannelByWatchInfo(nodeID int64, info *datapb.ChannelWatchInfo) *St
 	return c
 }
 
-func (c *StateChannel) TransitionOnSuccess() {
+func (c *StateChannel) TransitionOnSuccess(opID int64) {
+	if opID != c.Info.GetOpID() {
```
Comment (liliu-z) [inline internal/datacoord/channel.go:195]: Approve for now. But we definitely need refactor here. We need to
1. Return the Channel status on DN (Watching/Wachted/Releasing/Released/...)
2. Group all info for a given channel on DC to determine the next step

Comment (liliu-z) [reply internal/datacoord/channel.go:195]: @XuanYang-cn

**Pull request merged**